python/kfserving/kfserving/kfmodels/xgboost.py

```diff
@@ -16,34 +16,47 @@
 import xgboost as xgb
 from xgboost import XGBModel
 import os
-import numpy as np
-from typing import List, Dict
+from typing import Dict
 
 BOOSTER_FILE = "model.bst"
+DEFAULT_NTHREAD = 1
+
 
 class XGBoostModel(kfserving.KFModel):
-    def __init__(self, name: str, model_dir: str, nthread: int, booster: \
-        XGBModel = None):
+    def __init__(self, name: str, model_dir: str, nthread: int = DEFAULT_NTHREAD,
+                 booster: XGBModel = None):
         super().__init__(name)
         self.name = name
+        self.full_model_path = ""
         self.model_dir = model_dir
         self.nthread = nthread
         if not booster is None:
             self._booster = booster
             self.ready = True
 
-    def load(self):
+    def load_from_model_dir(self):
         model_file = os.path.join(
             kfserving.Storage.download(self.model_dir), BOOSTER_FILE)
-        self._booster = xgb.Booster(params={"nthread" : self.nthread},
+        self._booster = xgb.Booster(params={"nthread": self.nthread},
                                     model_file=model_file)
         self.ready = True
 
+    def load_from_full_model_path(self):
```

Comment:
Similar to the comment for sklearn model, should we just remove the load_from_full_model_path function? I don't see any reason to write code like below:
```
xgbModel. full_model_path = somePath
xgbModel. load_from_full_model_path()
```